solidity/contracts/middleware/libs/InterchainAccountMessage.sol

```diff
@@ -34,13 +42,15 @@ library InterchainAccountMessage {
     ) internal pure returns (bytes memory) {
         CallLib.Call[] memory _calls = new CallLib.Call[](1);
         _calls[0] = CallLib.build(_to, _value, _data);
-        return
-            abi.encode(
-                TypeCasts.addressToBytes32(_owner),
-                _ism,
-                _calls,
-                EMPTY_SALT // Salts are expected when decoding.
-            );
+
+        bytes memory prefix = abi.encodePacked(
+            MessageType.CALLS,
+            _owner.addressToBytes32(),
+            _ism,
+            EMPTY_SALT
```

Comment:
why is there a salt on the calls message type?

Reply:
We're using the salt to derive the ica, so we still need to include it, even it's zero.